### PR TITLE
Enable more linters esp. golint

### PR DESCRIPTION
**Pull request by guillaumerose**
Activate common linters. 
golint triggered lot of small changes. I find the code to be a little bit better now. Function names reads better.

### CODE REVIEW

File: pkg/crc/machine/hyperkit/driver_darwin.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/code-ready/machine/libmachine/drivers"
 )
 
-type hyperkitDriver struct {
+type Driver struct {
```
Comment (praveenkumar) [inline pkg/crc/machine/hyperkit/driver_darwin.go:11]: @guillaumerose Does golint also complain about type names like `libvirtDriver` or `hyperkitDriver` ?

Comment (guillaumerose) [reply pkg/crc/machine/hyperkit/driver_darwin.go:11]: Yes because the package name is the prefix of the struct name.

**Pull request merged**